pkg/etcdutil/etcdutil.go

```diff
@@ -292,5 +308,98 @@ func TakeAndSaveFullSnapshot(ctx context.Context, client client.MaintenanceClose
 	metrics.SnapshotDurationSeconds.With(prometheus.Labels{metrics.LabelKind: brtypes.SnapshotKindFull, metrics.LabelSucceeded: metrics.ValueSucceededTrue}).Observe(timeTaken.Seconds())
 	logger.Infof("Total time to save full snapshot: %f seconds.", timeTaken.Seconds())
 
+	defer func() {
+		if err := os.Remove(snapshotTempDBPath); err != nil {
+			logger.Warnf("failed to remove temporary full snapshot file: %v", err)
+		}
+		if err := snapshotData.Close(); err != nil {
+			logger.Warnf("failed to close snapshot data file: %v", err)
+		}
+	}()
```

Comment:
as discussed offline, deferred func is being used here to close the open resources, that's why it's important to put them in deferred func, so that resources get closed in case any failure in taking snapshot.

Reply:
Thanks. I'd missed seeing the error checks being done before. Sorry for bothering with this trivial comment.